### ISSUE TITLE
Redis reconnect

### ISSUE THREAD

**Issue by RicterZ** [yagop/telegram-bot#186]
The redis connection always interrupted after few mins.

    User Ricter Zheng marked read 1 outbox and 0 inbox messages   
     *** lua: .luarocks/share/lua/5.2/redis.lua:269: connection error: closed

**Comment by rockneurotiko** [comment]
Maybe is a problem of your redis installation?

I have tg-bot installed in a raspberry pi (with 256MB of RAM) and don't disconnect of the redis server, works well (right now 9 days up).

Some information about your system or how you have redis installed? :-)

**Comment by RicterZ** [comment]
Got. I will try to reinstall redis. 
:D

**Comment by RicterZ** [comment]
@rockneurotiko Hello, I reinstall redis-server, but it still interrupted.  
I installed redis by `apt-get` command, and my system is Ubuntu 14.04.

**Comment by psykomantis** [comment]
yes after 10 minutes of idle status my bot disconnect with the same error.

**Comment by quiquebg** [comment]
i have the same problem. but I find a temporary solution: close and open automatically every 5 minutes .

**Comment by rockneurotiko** [comment]
Are you guys executing with redis-server directly didn't you?
Try to install it properly following the steps that are described [here](http://redis.io/topics/quickstart) in the section `Installing Redis more properly`
I have redis installed like that and I don't have any problem :S
(also, don't install with apt-get, download [http://download.redis.io/redis-stable.tar.gz](http://download.redis.io/redis-stable.tar.gz) and follow the steps of the previous link :)

**Comment by Walkman100** [comment]
> also, don't install with apt-get

I installed with apt-get, never had a problem with it :/

**Comment by rockneurotiko** [comment]
@Walkman100 I always try to install from sources, so I thinked that maybe it was because of apt-get XD

**Comment by psykomantis** [comment]
yes my redis is working, maybe I will uninstall and compile the sources but I don't need to run my bot always so in my case it's unnecessary
